src/object/README.md

```diff
@@ -1,261 +1,145 @@
-
-- I/O requests against replicas are directly issued via DAOS client; there is no sequential guarantee on writes in the same epoch, and concurrent writes for a same object can arrive at different replicas in an arbitrary order.
-- Because there is no communication between servers in this way, there is no consistent guarantee if there are overlapped writes or KV updates in the same epoch. The DAOS server should detect overlapped updates in the same epoch, and return errors or warnings for the updates to the client. The only exception is multiple updates to the same extent or KV having the exactly same data. In this case, it is allowed because these updates could potentially be the resending requests.
+Client replication is the mode that it is synchronous and fully in the client
+stack, to provide high concurrency and low latency I/O for the upper layer.
+- I/O requests against replicas are directly issued via DAOS client; there is
+  no sequential guarantee on writes in the same epoch, and concurrent writes for
+  a same object can arrive at different replicas in an arbitrary order.
+- Because there is no communication between servers in this way, there is no
+  consistent guarantee if there are overlapped writes or KV updates in the same
+  epoch. The DAOS server should detect overlapped updates in the same epoch, and
+  return errors or warnings for the updates to the client. The only exception is
+  multiple updates to the same extent or KV having the exactly same data. In
+  this case, it is allowed because these updates could potentially be the
+  resending requests.
+
+Furthermore, when failure occurs, the client replication protocol still needs
+some extra mechanism to enforce consistency between replicas:
+- If the DAOS client can capture the failure, e.g. a target failed during I/O,
+  because the DAOS client can complete this I/O by switching to another target
+  and resubmitting request. At the meanwhile, the DAOS servers can rebuild the
+  missing replica in the background. Therefore, DAOS can still guarantee data
+  consistency between replicas. This process is transparent to DAOS user.
+- If DAOS cannot capture the failure, for example, the DAOS client itself
+  crashed before successfully updating all replicas so some replicas may lag
+  behind. Based on the current replication protocol, the DAOS servers cannot
+  detect the missing I/O requests, so DAOS cannot guarantee data consistency
+  between replicas. The upper layer stack has to either re-submit the
+  interrupted I/O requests to enforce data consistency between replicas, or
+  abort the epoch and rollback to the consistent status of the container.
+  For example, if an application linked to the DAOS client library crashed, then
+  it should either replay uncompleted I/O requests if it has log for those I/Os,
+  or abort the uncommitted epoch.
```

Comment:
These reasoning probably can be simplified, just say  " To keep the consistency of each replicas,  another I/O mode ...."